torch/masked/maskedtensor/binary.py

```diff
@@ -94,7 +94,10 @@ def _binary_helper(fn, args, kwargs, inplace):
     )
 
     args0_layout = data_args[0].layout
-    same_layout = args0_layout == data_args[1].layout
+    same_layout = (
+        (torch.is_tensor(data_args[1]) or is_masked_tensor(data_args[1])) and
```

Comment:
Why is this additional check necessary?

Reply:
This is necessary because there are cases where we do `MaskedTensor + float`, and `float`s don't have a dtype so that would error out